src/wallet.ts

```diff
@@ -256,6 +256,9 @@ export default class Wallet {
   transfer(input: {to: string, value: number, from?: string, nonce?: number, gas_price?: number}, isDryrun: boolean = false): Promise<any> {
     const address = this.getImpliedAddress(input.from);
     const toAddress = Ain.utils.toChecksumAddress(input.to);
+    if (!(input.value > 0)) {
```

Comment:
Is there any reason to check value is positive number? I know, If value is 0 or negative number, blockchain will throw error.

Reply:
Right, this saves user's tx fee though.

Reply:
Oh, if the transaction fails, does the user still have to pay the transaction fee?

Reply:
That's a good quesiton.
It depends on the conditions the transaction breaks.
If the're prechecks or skipFees = true, tx fee is not charged (see https://github.com/ainblockchain/ain-blockchain/blob/master/db/index.js#L1955).

Reply:
Thank you for the detailed explanation!